File: run_migration.mjs
import mysql from 'mysql2/promise';
import fs from 'fs';

async function run() {
  const dbUrl = process.env.DATABASE_URL;
  if (!dbUrl) {
    console.error('DATABASE_URL not set');
    process.exit(1);
  }
  const conn = await mysql.createConnection(dbUrl);
  const sql = fs.readFileSync('drizzle/migrations/featured_products.sql', 'utf8');
  const statements = sql.split(';').filter(s => s.trim() && !s.trim().startsWith('--'));
  for (const stmt of statements) {
    try {
      await conn.execute(stmt);
      console.log('OK:', stmt.substring(0, 80).replace(/\n/g, ' '));
    } catch (e) {
      console.error('ERR:', e.message.substring(0, 100));
    }
  }
  await conn.end();
  console.log('Migration done');
}
run();

File: run_migration.py
import mysql.connector
import os
import re
from urllib.parse import urlparse

database_url = os.environ.get('DATABASE_URL')
if not database_url:
    raise RuntimeError('DATABASE_URL is required')
parsed = urlparse(database_url)

conn = mysql.connector.connect(
    host=parsed.hostname,
    port=parsed.port or 3306,
    user=parsed.username,
    password=parsed.password,
    database=parsed.path.lstrip('/'),
    ssl_disabled=False,
    ssl_verify_cert=False,
    use_pure=True
)

cur = conn.cursor()

with open('drizzle/migrations/featured_products.sql', 'r') as f:
    sql = f.read()

# Split by semicolons, filter out comments and empty
statements = [s.strip() for s in sql.split(';') if s.strip() and not s.strip().startswith('--')]

for stmt in statements:
    try:
        cur.execute(stmt)
        conn.commit()
        print(f'OK: {stmt[:80].replace(chr(10), " ")}')
    except Exception as e:
        print(f'ERR: {str(e)[:100]}')

cur.close()
conn.close()
print('Migration done')
